### PR TITLE
STUD-69419 add flowchart interfaces

### PR DESCRIPTION
**Pull request by andrei-balint**
Define flowchart and flow nodes interfaces.
The purpose is to allow a designer host app to handle multiple flowchart implementations.

### CODE REVIEW

File: src/UiPath.Workflow.Runtime/Statements/FlowNode.cs [hunk 1]
```diff
@@ -60,4 +62,11 @@ internal void GetChildActivities(ICollection<Activity> children)
     }
 
     internal abstract void GetConnectedNodes(IList<FlowNode> connections);
+
+    IEnumerable<IFlowNode> IFlowNode.GetConnectedNodes()
+    {
+        var result = new List<FlowNode>();
+        GetConnectedNodes(result);
+        return result;
+    }
```
Comment (mihainradu) [inline src/UiPath.Workflow.Runtime/Statements/FlowNode.cs:71]: why?

Comment (andrei-balint) [reply src/UiPath.Workflow.Runtime/Statements/FlowNode.cs:71]: There is reflection code in Studio and I would like to take the opportunity to add a proper method.

Comment (mihainradu) [reply src/UiPath.Workflow.Runtime/Statements/FlowNode.cs:71]: as aa warning, this is internal and I would expect to be able to rename it

Comment (andrei-balint) [reply src/UiPath.Workflow.Runtime/Statements/FlowNode.cs:71]: unfortunately Studio is using it with reflection as there are no contracts available

**Comment by andrei-balint** [comment]
will be declared by design time assembly only

**Pull request closed without merging**